### PR TITLE
Add support for Oracle Cloud Infrastructure (OCI)

### PR DESCRIPTION
**Pull request by antonacio**
## Description

Kedro currently does not support reading and writing from and to Oracle Cloud Infrastructure (OCI) Object Storage filesystem, because the "oci://" protocol is not included in the list of supported cloud protocols.

## Development notes

Similarly to Azure Data Lake Storage Gen2 URIs, [OCI's Object Storage filesystem (ocifs)](https://ocifs.readthedocs.io/en/latest/) uses the @ syntax for I/O operations: "oci://bucket@namespace/path_to_file". 

Therefore, to make Kedro datasets work with OCIFS, I added "oci" to the `CLOUD_PROTOCOLS` list and included the same file path processing step done for Azure Data Lake Storage Gen2 URIs.

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [x] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

**Comment by antonacio** [comment]
Hey Nok, thanks for your response.

I just pushed a commit adding the test case for the OCI protocol. Regarding the OCI environment, the screenshots below show a couple of examples to illustrate the issue and the proposed solution -- let me know if makes sense to you.

P.S. The custom dataset here is just a copy of Kedro's `pandas.ParquetDataset` source code where I changed the imports to include the solution I propose in this PR.

oci-screenshot-write
oci-screenshot-read

**Review by noklam — APPROVED** [review]

**Comment by noklam** [comment]
Amazing, thank you for adding the test, this looks good to me!

**Review by ankatiyar — APPROVED** [review]

**Review by SajidAlamQB — APPROVED** [review]
Thank you, @antonacio!

**Comment by ankatiyar** [comment]
@antonacio Could you add this to the release notes?

File: kedro/io/core.py [hunk 1]
```diff
@@ -820,9 +821,11 @@ def _parse_filepath(filepath: str) -> dict[str, str]:
         host_with_port = parsed_path.netloc.rsplit("@", 1)[-1]
         host = host_with_port.rsplit(":", 1)[0]
         options["path"] = host + options["path"]
-        # Azure Data Lake Storage Gen2 URIs can store the container name in the
-        # 'username' field of a URL (@ syntax), so we need to add it to the path
-        if protocol == "abfss" and parsed_path.username:
+        # - Azure Data Lake Storage Gen2 URIs can store the container name in the
```
Comment (astrojuanlu) [inline kedro/io/core.py:824]: Wish we could contribute all these to `fsspec`

**Comment by antonacio** [comment]
> @antonacio Could you add this to the release notes?

@ankatiyar, done!

**Pull request merged**